### PR TITLE
Add FontAwesome icon support

### PR DESCRIPTION
**Pull request by jaredcwhite**
## Issue Overview

This PR addresses #2634 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR updates our `Icon` component so that it supports FontAwesome icons as well as the existing icons. You just import an icon directly from FontAwesome's React library, and supply that import to the `symbol` prop. For example:

```js
import { faForwardStep } from "@fortawesome/free-solid-svg-icons"

<Icon symbol={faForwardStep} size="large" fill="#2277CC" />
```

Because of the typing of the icon imports, the `Icon` component itself doesn't need to know anything about the particular icon set. So _any_ FontAwesome icons from free or paid sets can be used however needed in the consumer apps.

## How Can This Be Tested/Reviewed?

There are [FontAwesome icon stories](https://deploy-preview-2768--dev-storybook-bloom.netlify.app/?path=/story/icons-fontawesome-examples--icon-align-center) in Storybook, and there's also an [example of a Button](https://deploy-preview-2768--dev-storybook-bloom.netlify.app/?path=/story/actions-button--with-font-awesome-icon) using FontAwesome icons.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/pages/index.tsx [hunk 1]
```diff
@@ -67,6 +69,13 @@ export default function Home(props: IndexProps) {
         </AlertBox>
       )}
       <Hero title={heroTitle} buttonTitle={t("welcome.seeRentalListings")} buttonLink="/listings" />
+      <div style={{ color: "maroon" }}>
```
Comment (emilyjablonski) [inline sites/public/pages/index.tsx:72]: 🧹

File: ui-components/src/icons/Icon.tsx [hunk 2]
```diff
@@ -155,16 +159,31 @@ const Icon = (props: IconProps) => {
...
-  const SpecificIcon = IconMap[props.symbol]
+  const SpecificIcon =
+    typeof props.symbol === "string" ? (
+      IconMap[props.symbol as string]
+    ) : (
+      <FontAwesomeIcon icon={props.symbol} />
+    )
 
-  return (
+  console.info(typeof props.symbol === "string", SpecificIcon)
```
Comment (emilyjablonski) [inline ui-components/src/icons/Icon.tsx:169]: 🧹

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Just some cleanup needed

**Comment by jaredcwhite** [comment]
@emilyjablonski Yikes! Thanks for catching that 😅

**Review by emilyjablonski — APPROVED** [review]

**Comment by jaredcwhite** [comment]
> We have an NPM token that comes with our paid account. Do you need this for implementation?

@slowbot My initial impression is we'd want to keep any paid FA icons out of the open source core (because obviously they're not open source), and carefully discuss how we might want to incorporate them into specific jurisdiction installations. At any rate, I don't expect any changes are needed for this particular PR, so it's safe to merge in before making those decisions.

**Pull request merged**